pyomo/contrib/mindtpy/util.py

```diff
@@ -1285,3 +1286,28 @@ def create_utility_block(model, name, solve_data):
     yield
     if created_util_block:
         model.del_component(name)
+
+
+def copy_var_list_values(from_list, to_list, config,
+                         skip_stale=False, skip_fixed=True,
+                         ignore_integrality=False):
+    """Copy variable values from one list to another.
+
+    Rounds to Binary/Integer if necessary
+    Sets to zero for NonNegativeReals if necessary
+    """
+    if ignore_integrality:
+        deprecation_warning("The 'ignore_integrality' argument no longer "
+                            "has any functionality.", version="6.4.2")
+
+    if len(from_list) != len(to_list):
+        raise ValueError('The lengths of from_list and to_list do not match.')
+
+    for v_from, v_to in zip(from_list, to_list):
+        if v_from.name != v_to.name:
+            raise ValueError('The name of the two variables is not the same.')
```

Comment:
Why do you add this check? It certainly seems valuable for debugging, but do you need it always? Even if you want the sanity check, couldn't you do it only once, after you've created the utility blocks with the lists of variables on them? Unless I'm misunderstanding, this checks something that has been checked before in every iteration after the first.

In addition, this is duplicated code with the same function in `gdpopt/util.py`. If you do need this, I'd rather we make a `check_names` option in that method rather than putting the same code in two places.

Reply:
I agree with you.